packages/discord.js/src/structures/Message.js

```diff
@@ -418,6 +420,26 @@ class Message extends Base {
       this.poll ??= null;
     }
 
+    if (data.message_snapshots) {
+      /**
+       * The message associated with the message reference
+       * @type {Array<Message>}
+       */
+      this.messageSnapshots = data.message_snapshots.map(snapshot => {
+        const channel = this.client.channels.resolve(this.reference.channelId);
+        const snapshotData = {
+          ...snapshot.message,
+          id: this.reference.messageId,
+          channel_id: this.reference.channelId,
+          guild_id: this.reference.guildId,
+        };
+
+        return channel ? channel.messages._add(snapshotData) : new this.constructor(snapshotData);
+      });
+    } else {
+      this.messageSnapshots = data.message_snapshots?.slice() ?? [];
```

Comment:
If `data.message_snapshots` is missing, then what's the point of doing `data.message_snapshots?.slice()`?

Reply:
Whoops, got things mixed up. Cheers for catching that